### PR TITLE
Build ESM and CJS

### PR DESCRIPTION
**Pull request by LukeShay**
I was running in to problems with using ESM in my project. This fixes that error so users can import the type they need. This also moves joi to a peer dep because there is no need to have it as a dependency when the users are already importing it. This will allow for a wider range of joi versions to be used with this package.

### CODE REVIEW

**Comment by mrjono1** [comment]
Hi @LukeShay this is great!

**Comment by LukeShay** [comment]
> Hi @LukeShay this is great!

Thanks! CI should hopefully pass now.

**Comment by brodyd795** [comment]
Hey @mrjono1! What's the merge/release schedule typically look like on this repo? 

(Asking because I'm really hoping to consume this change in my project early this coming week if possible)

**Comment by mrjono1** [comment]
Hey I should be able to get this released in a day or so

File: package.json [hunk 1]
```diff
@@ -18,31 +19,40 @@
...
+    "@typescript-eslint/parser": "^5.10.1",
+    "eslint": "^8.7.0",
     "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-prettier": "^4.0.0",
-    "jest": "^27.1.0",
-    "prettier": "^2.2.1",
-    "ts-jest": "^27.0.5",
-    "typescript": "4.5.4"
+    "jest": "^27.4.7",
+    "joi": "^17.4.0",
```
Comment (mrjono1) [inline package.json:49]: I think we need to remove this line
```suggestion
```

Comment (mrjono1) [reply package.json:49]: I have Joi 17.3.0 in another project and I'm getting Joi version conflicts 
`Error: Cannot mix different versions of joi schemas`
but when its removed it cannot find Joi, am I misunderstanding peer dependencies?

**Comment by mrjono1** [comment]
Released https://github.com/mrjono1/joi-to-typescript/releases/tag/3.0.0 :) thankyou for your work

**Comment by mrjono1** [comment]
npm has ignored many of the files im figuring out why

**Comment by mrjono1** [comment]
I have fixed the publish https://github.com/mrjono1/joi-to-typescript/issues/187 but I am having some trouble with using the ESM changes with my example project for some reason, it was working earlier let me know how you go

**Comment by mrjono1** [comment]
https://github.com/mrjono1/joi-to-typescript/issues/190

**Comment by LukeShay** [comment]
Thanks! I will do some troubleshooting when I get the chance. I linked it locally with a project I had to verify it works with ESM when I made the PR.

**Comment by mrjono1** [comment]
I did the same and it worked great, now having a few issues I'll try again.
Is it all working for you?

**Comment by brodyd795** [comment]
> I did the same and it worked great, now having a few issues I'll try again. Is it all working for you?

I haven't done a ton of testing with it so far, but I tried out the example in my project locally, and all's working well! Thanks for getting this to the finish line, @mrjono1 and @LukeShay!

**Pull request merged**